### PR TITLE
Restore prop `imageRequest` in `CrystalToolkitScene.tsx`

### PR DESCRIPTION
**Pull request by janosh**
Removed for unknown reasons in https://github.com/materialsproject/mp-react-components/commit/798a1979d78da8ce5ae28854ef624a61c406a080.

This broke the [CrystalToolkit script for programmatic screenshot generation](https://github.com/materialsproject/crystaltoolkit/blob/c4ff9e3b2a826eaa2ffff96eecc4c619bf466a3b/crystal_toolkit/apps/examples/write_structure_screenshot_to_file.py).

### CODE REVIEW

**Comment by janosh** [comment]
@mkhorton If you have time, would be great to get your eyes on this in case I missed any pieces of code needed to get the CTK script working again.

**Comment by mkhorton** [comment]
Thanks. I’ll take a look. I’m actually in the middle of a fair bit of improvements/refactoring of Crystal Toolkit so a well-timed ask!

**Comment by mkhorton** [comment]
Ok, I left a comment on this elsewhere but I'm unable to find it now.

Tl;dr, I think this PR is good. We should make sure to keep this prop.

This part of the code got quite messy over the years as different individuals edited it, it's a little overdue for a cleanup and more documentation. Open to suggestions for how we could better structure the code.

## More Context

The most important point to understand that we have two methods of triggering a download of an image or Three.js scene export:

* **Direct.** Within the React component, we export the current canvas to a png, or the current scene to DAE etc., and trigger a download of the png client side.
  * Pros: faster for the client, more efficient.
  * Cons: no way to intercept this image server side.

* **Indirect.** Set an associated `prop` with the contents of the file.
  * Pros: Can be used to save exported files server-side. This also allows for a feature I've wanted to add for a long time, which is to embed the legend directly into the image.
  * Cons: Requires a round-trip to the server. Also I'm uncertain if storing this kind of data in a prop might lead to some kind of performance issue, or limitation, especially for people on poorer Internet conections.

Then we have the question of how to trigger this request, which is this `imageRequest` prop. If we are using the "direct" method, this is strictly not required, but is useful for the "indirect" method and/or when scripting is required.

*Longer-term*, I am still hoping a good solution exists which allows us to render server-side without a browser, this might be:

* Using the Three.js renderer, but with something like `headless-gl`.
* Using an alternative renderer, and exporting the `Scene` to another program to render, e.g. Blender. There's another Python library which I can't find now which does exports to several packages; I probably would've used this library if I'd been aware of it when Crystal Toolkit was first developed.

**Comment by mkhorton** [comment]
Also, I hope it's ok, I appended a commit to this PR for an easy-addition of adding an additional `glb` export type.

It's outside the scope of this PR, but the main time required is to do the release cycle, so it seemed like a good opportunity to add it; it's useful for embedding in presentations etc.

**Comment by janosh** [comment]
> Also, I hope it's ok, I appended a commit to this PR for an easy-addition of adding an additional glb export type.

Of course, thanks @mkhorton!
